WalletWasabi/WabiSabi/Client/GreedyDecomposer.cs

```diff
@@ -11,33 +11,24 @@ public class GreedyDecomposer
 		public GreedyDecomposer(IEnumerable<Money> denominations)
 		{
 			Denominations = denominations.Any()
-				? denominations.OrderByDescending(x => x).ToImmutableList()
+				? denominations.OrderByDescending(x => x).ToImmutableArray()
 				: throw new ArgumentException($"Argument {nameof(denominations)} has no elements");
 		}
 
-		public ImmutableList<Money> Denominations { get; }
+		public ImmutableArray<Money> Denominations { get; }
 
-		public IEnumerable<Money> Decompose(Money amount)
+		public IEnumerable<Money> Decompose(Money amount, Money costPerOutput)
```

Comment:
The decomposer has not idea of the cost per outputs because it doesn't have the scriptPubKeys. This assumes there will always be only one type of outputs.

Reply:
i think that's fine since the decomposition is only for self spend outputs, which should have a uniform type